src/main/scala/mesosphere/marathon/api/akkahttp/v2/TasksController.scala

```diff
@@ -62,6 +70,53 @@ class TasksController(
     }
   }
 
+  private def deleteTasks()(implicit identity: Identity): Route = {
+    def doKill(force: Boolean, scale: Boolean, wipe: Boolean, tasksIdToAppId: Map[Id, PathId]) = {
+      def isAuthorized(tasksIdToAppId: Map[Id, PathId]): Boolean = tasksIdToAppId.values.exists(id => !authorizer.isAuthorized(identity, UpdateRunSpec, id))
+
+      // TODO authorization check is performed twice - in the taskKiller and also here, this one must remain
+      if (isAuthorized(tasksIdToAppId)) {
+        reject(AuthDirectives.NotAuthorized(HttpPluginFacade.response(authorizer.handleNotAuthorized(identity, _))))
+      } else {
+        val maybeInstances: Future[Iterable[Option[Instance]]] = Future.sequence(tasksIdToAppId.view
+          .map { case (taskId, _) => instanceTracker.instancesBySpec.map(_.instance(taskId)) })
+        val tasksByAppId: Future[Map[PathId, Seq[Instance]]] = maybeInstances.map(i => i.flatten
+          .groupBy(instance => instance.instanceId.runSpecId)
+          .map { case (appId, instances) => appId -> instances.to[Seq] }(collection.breakOut))
+
+        if (scale) {
+          onSuccess(killAndScale(tasksByAppId, force)) { deploymentResult =>
```

Comment:
What type is `deploymentResult`?

Reply:
DeploymentResult :) that is derived from DeploymentPlan in killAndScale method